server/src/main/java/io/spine/examples/kanban/server/board/BoardAggregate.java

```diff
@@ -176,8 +173,134 @@ private void apply(ColumnPlaced e) {
 
     @Apply
     private void apply(ColumnMovedOnBoard e) {
-        builder().removeColumn(e.getFrom().zeroBasedIndex())
-                 .addColumn(e.getTo().zeroBasedIndex(), e.getColumn());
+        int index = state().getColumnList().indexOf(e.getColumn());
+        int newIndex = e.getTo().zeroBasedIndex();
+        if (index != newIndex) {
+            builder().removeColumn(index).addColumn(newIndex, e.getColumn());
+        }
+    }
+
+    /**
+     * Moves the column to the desired position and shifts all columns on the
+     * way to fill the void left by the column.
+     */
+    @Assign
+    Iterable<ColumnMovedOnBoard> handle(MoveColumn c) throws ColumnCannotBeMoved {
+        return new ImmutableList.Builder<ColumnMovedOnBoard>()
+                .add(moveColumn(c))
+                .addAll(shiftColumns(c))
+                .build();
+    }
+
+    private ColumnMovedOnBoard moveColumn(MoveColumn c) throws ColumnCannotBeMoved {
+        if (!canBeMoved(c.getColumn(), c.getFrom(), c.getTo())) {
+            throw ColumnCannotBeMoved
+                    .newBuilder()
+                    .setColumn(c.getColumn())
+                    .setFrom(c.getFrom())
+                    .setTo(c.getTo())
+                    .build();
+        }
+
+        return ColumnMovedOnBoard
+                .newBuilder()
+                .setColumn(c.getColumn())
+                .setBoard(c.getBoard())
+                .setFrom(c.getFrom())
+                .setTo(c.getTo())
+                .vBuild();
+    }
+
+    private boolean canBeMoved(ColumnId column, ColumnPosition from, ColumnPosition to) {
+        return from.isValid() &&
+                to.isValid() &&
+                isTotalActual(from) &&
+                isTotalActual(to) &&
+                isIndexActual(column, from) &&
+                !from.equals(to);
+    }
+
+    /**
+     * Checks whether the total number of columns is coherent with the state.
+     *
+     * @return {@code true} if the total number of columns is coherent with the state.
```

Comment:
We don't put periods at the end of `@return` docs. See [this example](https://github.com/SpineEventEngine/base/blob/e6ec34b42564ff8cc4c61605bd1d5c008b573de4/base/src/main/java/io/spine/protobuf/AnyPacker.java#L68) Spine codebase, or any other.